### PR TITLE
chore: PoC - Schema code generation - Supporting nested attributes 

### PR DESCRIPTION
**Pull request by AgustinBettati**
## Description

Link to any related issue(s): CLOUDP-278462

Supports `SingleNestedAttribute`, `ListNestedAttribute`, `MapNestedAttribute`.

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals I have added appropriate changelog entries.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: tools/codegen/codespec/model.go [hunk 1]
```diff
@@ -71,9 +71,6 @@ type MapNestedAttribute struct {
 type NumberAttribute struct {
 	Default *CustomDefault
 }
-type ObjectAttribute struct {
-	Default *CustomDefault
-}
```
Comment (AgustinBettati) [inline tools/codegen/codespec/model.go:76]: leftover from previous PR

File: tools/codegen/codespec/model.go [hunk 2]
```diff
@@ -83,8 +80,8 @@ type SetNestedAttribute struct {
 	NestedObject NestedAttributeObject
 }
 type SingleNestedAttribute struct {
-	Default    *CustomDefault
-	Attributes Attributes
+	Default      *CustomDefault
+	NestedObject NestedAttributeObject
```
Comment (AgustinBettati) [inline tools/codegen/codespec/model.go:84]: To be consistent with other nested attribute models (List and Map)

**Review by EspenAlbert — APPROVED** [review]
LGTM!

**Review by maastha — APPROVED** [review]

File: tools/codegen/codespec/model.go (hunk 2 above)
Comment (maastha) [reply tools/codegen/codespec/model.go:84]: In Tf schemas, `SingleNestedAttribute` actually define `Attributes: map[string]schema.Attribute{}` and not NestedObject
Do you still want to use NestedObject here?

Comment (AgustinBettati) [reply tools/codegen/codespec/model.go:84]: Allowed me to handle 3 nested cases in a common way handling NestedAttributeObject. Its a small detail but found it more clear.

**Pull request merged**